jupyterlab_git/git.py

```diff
@@ -16,7 +18,14 @@
 # See https://git-scm.com/docs/git-config#_syntax for git var syntax
 CONFIG_PATTERN = re.compile(r"(?:^|\n)([\w\-\.]+)\=")
 DEFAULT_REMOTE_NAME = "origin"
+# Ensure on NFS or similar, that we give the .git/index.lock time to be removed
+MAX_WAIT_FOR_LOCK_S = 5
+# How often should we check for the lock above to be free?
+CHECK_LOCK_INTERVAL_S = 0.1
+# How long to wait to be executed or finished your execution before timing out
+MAX_WAIT_FOR_EXECUTE_S = 20
```

Comment:
There is an inconsistency here. The waiting time for the lock should be greater than the one for execution. Otherwise, for a long run other calls will most likely be dropped due to a failure in acquiring the lock.